# Jane Street Market Prediction
[Kaggle Jane Street Market Prediction](https://www.kaggle.com/c/jane-street-market-prediction/overview)


In [1]:
import pandas as pd
import numpy as np
import datatable as dt
X = dt.fread('jane-street-market-prediction/train.csv').to_pandas()

## Calculate utility score of training set where `action` = 1 for all `resp` > 0
[training set](https://www.kaggle.com/c/jane-street-market-prediction/data?select=train.csv)

### Utility score
224162.2681796676

## Split training set and fit tree

In [2]:
Xt = X[X.date >= 460]
yt = np.where((Xt.resp > 0), 1, 0)
X = X[(X.date) > 85 & (X.date < 460)]
y = np.where((X.resp > 0), 1, 0)
X.drop(['date', 'resp_1', 'resp_2', 'resp_3', 'resp_4', 'resp', 'ts_id'], axis=1, inplace=True)
Xt.drop(['date', 'resp_1', 'resp_2', 'resp_3', 'resp_4', 'resp', 'ts_id'], axis=1, inplace=True)

In [3]:
from sklearn.experimental import enable_hist_gradient_boosting
from sklearn.ensemble import HistGradientBoostingClassifier
clf = HistGradientBoostingClassifier(max_iter=100, verbose=1, random_state=1000).fit(X, y)

Binning 2.241 GB of training data: 24.128 s
Binning 0.249 GB of validation data: 1.756 s
Fitting gradient boosted rounds:
[1/100] 1 tree, 31 leaves, max depth = 11, train loss: 0.69274, val loss: 0.69275, in 2.281s
[2/100] 1 tree, 31 leaves, max depth = 8, train loss: 0.69241, val loss: 0.69244, in 2.008s
[3/100] 1 tree, 31 leaves, max depth = 9, train loss: 0.69211, val loss: 0.69214, in 2.067s
[4/100] 1 tree, 31 leaves, max depth = 12, train loss: 0.69182, val loss: 0.69186, in 2.208s
[5/100] 1 tree, 31 leaves, max depth = 9, train loss: 0.69150, val loss: 0.69157, in 2.310s
[6/100] 1 tree, 31 leaves, max depth = 8, train loss: 0.69127, val loss: 0.69136, in 2.111s
[7/100] 1 tree, 31 leaves, max depth = 9, train loss: 0.69107, val loss: 0.69117, in 2.110s
[8/100] 1 tree, 31 leaves, max depth = 11, train loss: 0.69082, val loss: 0.69092, in 2.867s
[9/100] 1 tree, 31 leaves, max depth = 9, train loss: 0.69064, val loss: 0.69076, in 2.195s
[10/100] 1 tree, 31 leaves, max depth = 9, trai

In [4]:
del X, y

In [5]:
clf.score(Xt, yt)

0.545623801193999

### max_iter=100, verbose=1, random_state=None
0.5455018688476925

### max_iter=100, verbose=1, random_state=1
0.545544076198337

### max_iter=100, verbose=1, random_state=100
0.5440808880426575

### max_iter=100, verbose=1, random_state=1000
0.545623801193999

In [ ]:
X = pd.concat([example_sample_submission.set_index('ts_id'), example_test.set_index('ts_id')], axis=1)

In [ ]:
X

In [ ]:
X['pj'] = X.weight * X.action
pi = X.groupby(['date']).pj.sum()
t = pi.sum()/((pi**2).sum()**0.5) * (250/pi.count())**0.5
u = min(max(t, 0), 6) * pi.sum()
print(u)

In [ ]:
X['pj'] = X.weight * np.where((X.resp > 0), X.resp, 0)
pi = X.groupby(['date']).pj.sum()
t = pi.sum()/((pi**2).sum()**0.5) * (250/pi.count())**0.5
u = min(max(t, 0), 6) * pi.sum()
print(u)